src/utils_gen/utils.jl

```diff
@@ -270,3 +270,288 @@ function save_results(model::Model; directory = joinpath(pwd(), "csv_files"))
+    types = []
+    for name ∈ names(modul)
+        if isdefined(modul, name) && (
+            getfield(modul, name) isa DataType || (
+                getfield(modul, name) isa UnionAll &&
+                Base.unwrap_unionall(getfield(modul, name)) isa DataType
+            )
+        )
+            push!(types, name)
+        end
+    end
+    return types
+end
+
+function get_types(moduls::Vector{Module})
+    types=[]
+    for modul ∈ moduls
+        append!(types, get_types(modul))
+    end
+    return types
+end
+
+function get_types(pkg::Union{String,Symbol})
+    return get_types(getfield(Main, Symbol(pkg)))
+end
+
+function get_types(pkgs::Union{Vector{<:Union{String,Symbol}},Set{<:Union{String,Symbol}}})
+    types = []
+    for pkg ∈ pkgs
+        append!(types, get_types(pkg))
+    end
+    return types
+end
+
+"""
+    get_supertypes(input) -> Dict{Symbol, Vector{Type}}
+
+Retrieves the supertypes of all defined types from modules or packages.
+
+# Method Overloads
+
+- `get_supertypes(modul::Module)`:
+  Returns a dictionary mapping type names to their supertypes from the given module.
+
+- `get_supertypes(moduls::Vector{Module})`:
+  Merges and returns supertypes from multiple modules.
+
+- `get_supertypes(pkg::Union{String, Symbol})`:
+  Converts the package name to a module (via `Main`) and returns its type supertypes.
+
+- `get_supertypes(pkgs::Union{Vector{<:Union{String, Symbol}}, Set{<:Union{String, Symbol}}})`:
+  Merges and returns supertypes from multiple packages via their names.
+
+# Arguments
+- `input`: Can be a single module, a vector of modules, a single package name (as `String` or `Symbol`), or a collection of package names.
+
+# Returns
+- `Dict{Symbol, Vector{Type}}`: A dictionary where each key is a type name and the value is a vector of its supertypes.
+
+# Description
+This set of functions helps extract the inheritance hierarchy of types defined in Julia modules or packages. It filters out non-type bindings and collects supertypes using `supertypes`.
+
+"""
+function get_supertypes(modul::Module)
+    types=Dict()
+    for name ∈ names(modul)
+        if isdefined(modul, name) && (
+            getfield(modul, name) isa DataType || (
+                getfield(modul, name) isa UnionAll &&
+                Base.unwrap_unionall(getfield(modul, name)) isa DataType
+            )
+        )
+            types[name] = supertypes(getfield(modul, name))
+        end
+    end
+    return types
+end
+
+function get_supertypes(moduls::Vector{Module})
+    types=Dict()
+    for modul ∈ moduls
+        merge!(types, get_supertypes(modul))
+    end
+    return types
+end
+
+function get_supertypes(pkg::Union{String,Symbol})
+    return get_supertypes(getfield(Main, Symbol(pkg)))
+end
+
+function get_supertypes(
+    pkgs::Union{Vector{<:Union{String,Symbol}},Set{<:Union{String,Symbol}}},
+)
+    types = Dict()
+    for pkg ∈ pkgs
+        merge!(types, get_supertypes(pkg))
+    end
+    return types
+end
```

Comment:
Consider compactifying:
```suggestion
get_supertypes(moduls::Vector{Module}) =
    merge!(Dict(), (get_supertypes(m) for m ∈ moduls)...)
get_supertypes(pkg::Union{String,Symbol}) = get_supertypes(getfield(Main, Symbol(pkg)))
get_supertypes(pkgs::Union{Vector{<:Union{String,Symbol}},Set{<:Union{String,Symbol}}}) =
    merge!(Dict(), (get_supertypes(pkg) for pkg ∈ pkgs)...)
```